src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/CaseLinkService.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.ccd.domain.service.casedeletion;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseLinkEntity;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseLinkRepository;
+import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsRepository;
+import uk.gov.hmcts.ccd.data.casedetails.DefaultCaseDetailsRepository;
+import uk.gov.hmcts.ccd.domain.model.casedeletion.CaseLink;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+@Slf4j
+@Service
+public class CaseLinkService {
+
+    private final CaseDetailsRepository caseDetailsRepository;
+    private final CaseLinkRepository caseLinkRepository;
+    private final CaseLinkMapper caseLinkMapper;
+
+    @Inject
+    public CaseLinkService(CaseLinkRepository caseLinkRepository,
+                           CaseLinkMapper caseLinkMapper,
+                           @Qualifier(DefaultCaseDetailsRepository.QUALIFIER)
+                                   CaseDetailsRepository caseDetailsRepository) {
+        this.caseLinkRepository = caseLinkRepository;
+        this.caseLinkMapper = caseLinkMapper;
+        this.caseDetailsRepository = caseDetailsRepository;
+    }
+
+    public void updateCaseLinks(Long caseId,
+                                String caseTypeId,
+                                List<String> preCallbackCaseLinks,
+                                List<String> postCallbackCaseLinks) {
+        deleteRemovedCaseLinks(caseId, preCallbackCaseLinks, postCallbackCaseLinks);
+        insertNewCaseLinks(caseId, caseTypeId, preCallbackCaseLinks, postCallbackCaseLinks);
+    }
+
+    public void createCaseLinks(Long caseReference, String caseTypeId, List<String> caseLinks) {
+        caseLinks.stream()
+            .filter(caseLinkString -> caseLinkString != null && !caseLinkString.isEmpty())
+            .forEach(caseLink -> {
+                caseLinkRepository.insertUsingCaseReferenceLinkedCaseReferenceAndCaseTypeId(caseReference,
+                    Long.parseLong(caseLink),
+                    caseTypeId);
+                log.debug("inserted case link with id {}, linkedCaseId {} and caseType {}",
+                    caseReference, caseLink, caseTypeId);
+            });
+
+    }
+
+    private void deleteRemovedCaseLinks(Long caseId,
+                                        List<String> preCallbackData,
+                                        List<String> postCallbackData) {
```

Comment:
Can we rename all three case fields to highlight that this function is expecting references.

Reply:
Done in [790b9eb](https://github.com/hmcts/ccd-data-store-api/pull/1719/commits/790b9ebd63a212b252fd036ceb8cc804f1679593)